staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go

```diff
@@ -185,6 +188,15 @@ var _ DiscoveryAggregationController = &discoveryManager{}
 
 func NewDiscoveryManager(
 	target discoveryendpoint.ResourceManager,
+) DiscoveryAggregationController {
+	return NewLegacyDiscoveryManager(target, nil)
+}
+
+// NewLegacyDiscoveryManager creates a new DiscoveryAggregationController that
+// supports registering the core API group into a separate legacy ResourceManager.
+func NewLegacyDiscoveryManager(
+	target discoveryendpoint.ResourceManager,
+	legacy discoveryendpoint.ResourceManager,
```

Comment:
why not just wrap the resource manager and intercepts the legacy group calls? Wouldn't that give you the same ability?